### DISCUSSION TITLE
Proposal to find a zod alternative

### DISCUSSION THREAD

**Discussion by gnapse** [epicweb-dev/epic-stack#722] [General]
I've suspected of this myself in bigger projects, but now I see I'm not alone:

> Zod is really nice but it has zero memoization. It loves bound instance functions. z.number().int().positive() allocates 81 functions (4 kilobytes of memory).
>
> Is there an alternative with a similar API that is better about initialization performance?
>
> [source](https://twitter.com/sebmck/status/1787188422433497188)

Is it worth it to consider replacing zod with something else? In the tweet comments they mention some alternatives, but I haven't used them, so not sure if they're better both from the features POV and the performance aspect.

**Comment by kentcdodds** [comment]
I welcome an [example](https://github.com/epicweb-dev/epic-stack/blob/main/docs/examples.md) demonstrating using something else.

I'd love Zod to be more optimized, but until someone can demonstrate an improvement with something else (or until it becomes enough of a problem for me that I try) then Zod will remain.
